atom/browser/api/atom_api_app.cc

```diff
@@ -225,6 +226,42 @@ void App::OnLogin(LoginHandler* login_handler) {
     login_handler->CancelAuth();
 }
 
+bool App::CanCreateWindow(const GURL& opener_url,
+                     const GURL& opener_top_level_frame_url,
+                     const GURL& source_origin,
+                     WindowContainerType container_type,
+                     const GURL& target_url,
+                     const content::Referrer& referrer,
+                     WindowOpenDisposition disposition,
+                     const blink::WebWindowFeatures& features,
+                     bool user_gesture,
+                     bool opener_suppressed,
+                     content::ResourceContext* context,
+                     int render_process_id,
+                     int opener_render_view_id,
+                     int opener_render_frame_id,
+                     bool* no_javascript_access) {
+  // just a reminder that we are on the IO thread
+  // and need to be careful about v8 isolate usage
+  DCHECK_CURRENTLY_ON(content::BrowserThread::IO);
+
+  *no_javascript_access = false;
+
+  if (container_type == WINDOW_CONTAINER_TYPE_BACKGROUND) {
+    return true;
+  }
+
+  // this will override allowpopups
+  // we need some way to integerate
+  // it so we can turn popup blocking
+  // off if desired
+  if (!user_gesture) {
```

Comment:
Super minor nit: capital on first character of a comment and you can probably make this comment only span 2 lines and still stay under the char count.